### DISCUSSION TITLE
SOURCES.md revamp

### DISCUSSION THREAD

**Discussion by chunshek** [breezy-weather/breezy-weather#1411] [Show and tell]
**TL;DR:** I made a more "user-friendly" [mock-up](https://github.com/chunshek/breezy-weather/blob/docs/docs/SOURCES.md) of `SOURCES.md`.

---

I was going through [`SOURCES.md`](https://github.com/breezy-weather/breezy-weather/blob/main/docs/SOURCES.md) earlier and thought: this must be very confusing for a first-time user.

Imagine if you were a regular Canadian (or any other country really) – not a weather nerd, just someone who wants to check out some "open source software" that you had heard your nerdy friends talking about. They told you about Breezy Weather. So you came to the Breezy Weather GitHub page. You saw a bullet point that said **"Multiple weather sources (comparison)"**. You followed the link to [`SOURCES.md`](https://github.com/breezy-weather/breezy-weather/blob/main/docs/SOURCES.md). You might not know what the acronym ECCC stood for, or that it referred to the Canadian agency responsible for weather forecast. You saw a page _full_ of tables, filled with acronyms you didn't recognize. Most likely, your eyes glazed over at this point. You decided that you couldn't quickly find "Canadian weather" in this "open source software" and left.

Fine! Let's say that you had a bit more patience and paid a bit more attention. You saw "ECCC" and "Canada" in the same column of the first table, and put two and two together. But now, you see ECCC scattered across 6 different tables from the top to the bottom of the page. There were even **two separate but (almost) identical** tables that talked about the availability of "Current, Air Quality, Pollen, Precipitation, Alert, Normals"! You hadn't even started using the software. You wouldn't know what they meant.

In each table, Canada was being compared against China and Sweden and Ireland and a bunch of other countries – which, as a regular Canadian user, you didn't really care what those folks got out of _their_ weather sources. You just wanted to know what _Canadian weather_ you could get out of this software. You didn't want to scroll up and down through 6 tables to piece all the relevant information together. That was just too much mental work for a regular user.

Don't get me wrong: I'm a geek. I eat mental work for breakfast. But I'm aware most people don't.

I would like to propose a new format for `SOURCES.md` page. Rather than breaking down the sources from a feature/attribute-first perspective, we reorganize the information by country first. If I were a user from Country X, when I opened `SOURCES.md`, I could see my national source listed in a summary table. When I clicked on that link, it would show me what the source offers. If a particular source required an API key, a link would take me to a page that explains step-by-step how I could get one.

I created a [mock-up](https://github.com/chunshek/breezy-weather/blob/docs/docs/SOURCES.md) to illustrate what it _could_ look like. It's still a rough draft, neither final nor complete. I'll be happy to take suggestions and polish it further (or drop it).

**Comment by papjul** [comment]
Looks neat. I was thinking of revamping it as part of #169 but this keeps getting pushed further…

Other improvements I see:
- Developer side: dealing with new columns was getting harder, now it's much easier ;)
- I think it could be easier to give instructions in other languages than English. Let's say my app is set up in French, and I don't understand well English. We could have instructions in French for France-originating sources, because it can be a bit intimidating to tap the Help button and getting instructions only in English

Regressions I see:
- The emojis. They look nice, but they are not understandable at first. You need to look for the legend (which is not particularly visible), then expand it and cross check with the table. Not sure if that's something easy to do from a smartphone where display size doesn't allow to see both the table and the legend at the same time.

I will review it more in details and get back to you with some improvements idea, but it's a very good starting point ;)
